tools/ci/run_tc.py

```diff
@@ -262,13 +266,60 @@ def setup_environment(args):
     if args.oom_killer:
         start_userspace_oom_killer()
 
-    if args.checkout:
-        checkout_revision(args.checkout)
 
+def setup_repository(args):
+    is_pr = os.environ.get("GITHUB_PULL_REQUEST", "false") != "false"
+
+    # Initially task_head points at the same commit as the ref we want to test
+    # However that may not be the same commit as we actually want to test if
+    # the branch changed since the decision task ran.
```

Comment:
So this is where I think we can insert just a sprinkle of the explanation you gave me above. Something like:

```
    # Initially 'task_head' points to the same commit as the ref we want to test.
    # However it may no longer be the commit we actually want to test if the
    # branch has changed since the decision task ran. The branch may have
    # changed because someone has pushed more commits (either to the PR
    # or later commits to the branch), or because someone has altered the
    # base branch for the PR.
    #
    # We take a different approach depending on whether this is a PR or a
    # push to a branch. ... [the rest here]
```